File: src/main/java/me/kuku/yuq/job/NetEaseJob.java
package me.kuku.yuq.job;

import com.IceCreamQAQ.Yu.annotation.Cron;
import com.IceCreamQAQ.Yu.annotation.JobCenter;
import me.kuku.yuq.entity.NetEaseEntity;
import me.kuku.yuq.entity.NetEaseService;
import me.kuku.yuq.logic.NetEaseLogic;

import javax.inject.Inject;
import java.util.List;

@JobCenter
public class NetEaseJob {

	@Inject
	private NetEaseService netEaseService;
	@Inject
	private NetEaseLogic netEaseLogic;

	@Cron("At::d::08:06:32")
	public void sign(){
		List<NetEaseEntity> list = netEaseService.findAll();
		for (NetEaseEntity netEaseEntity : list) {
			try {
				netEaseLogic.sign(netEaseEntity);
				netEaseLogic.listeningVolume(netEaseEntity);
				netEaseLogic.musicianSign(netEaseEntity);
			} catch (Exception e) {
				e.printStackTrace();
			}
		}
	}

	@Cron("1h")
	public void musicianSign(){
		List<NetEaseEntity> list = netEaseService.findAll();
		for (NetEaseEntity netEaseEntity : list) {
			try {
				netEaseLogic.musicianSign(netEaseEntity);
			}catch (Exception e){
				e.printStackTrace();
			}
		}
	}

}

File: src/main/kotlin/me/kuku/yuq/job/NetEaseJob.kt
package me.kuku.yuq.job

import com.IceCreamQAQ.Yu.annotation.Cron
import com.IceCreamQAQ.Yu.annotation.JobCenter
import kotlinx.coroutines.delay
import me.kuku.yuq.entity.NetEaseService
import me.kuku.yuq.entity.Status
import me.kuku.yuq.logic.NetEaseLogic
import org.springframework.stereotype.Component

@JobCenter
@Component
class NetEaseJob (
    private val netEaseService: NetEaseService
) {


    @Cron("07:12")
    suspend fun sign() {
        val list = netEaseService.findAll().filter { it.config.sign == Status.ON }
        for (netEaseEntity in list) {
            kotlin.runCatching {
                delay(3000)
                NetEaseLogic.sign(netEaseEntity)
                delay(3000)
                NetEaseLogic.listenMusic(netEaseEntity)
            }
        }
    }

    @Cron("08:32")
    suspend fun musicianSign() {
        val list = netEaseService.findAll().filter { it.config.musicianSign == Status.ON }
        for (netEaseEntity in list) {
            kotlin.runCatching {
                for (i in 0..1) {
                    delay(3000)
                    NetEaseLogic.musicianSign(netEaseEntity)
                    delay(3000)
                    NetEaseLogic.publish(netEaseEntity)
                    delay(3000)
                    NetEaseLogic.publishMLog(netEaseEntity)
                }
            }
        }
    }

}